adr/adr-responses-api.md

```diff
@@ -0,0 +1,52 @@
+
+* **State Management:** Maintaining the conversation history (user messages, AI responses, tool outputs).
+* **Agentic Loop:** Orchestrating the turn-by-turn interaction with the Llama Stack Responses API. This involves:
+    1. Formatting the current state (chat history, available tools) into a request.
+    2. Sending the request to the API endpoint.
+    3. Parsing the API response to distinguish between textual replies and tool call requests.
+    4. Executing any requested tool calls using client-side or remote tooling.
+    5. Formatting the tool execution results.
+    6. Appending the tool results to the history and repeating the loop until a final answer is generated or a stop condition is met.
+* **Tool Dispatching:** Managing the registration and invocation of available tools on the client side.
+
+The decision of what form this client-side Agent construct will take will be addressed in a subsequent ADR. Some proposals include adding the functionality to the official Llama Stack client library, creating a new SDK maintained by Red Hat (perhaps as part of OpenDataHub), providing a container image with a simple HTTP API, and adopting a pre-existing SDK such as LangChain. That ADR will cover the pros and cons of these options and propose a final resolution.
+
+## Status
+
+Proposed
+
+## Consequences
+
+### Positive
+
+* **Strategic Alignment & Future-Proofing:** By building on the OpenAI-compatible Responses API, we align with the industry standard and the strategic direction of Llama Stack. This de-risks our platform from the likely deprecation of the older Agents API.
+* **Ecosystem Compatibility:** Our agentic capability will be natively compatible with the vast ecosystem of tools, libraries, and monitoring solutions that are built to interface with the OpenAI API format. This accelerates both internal development and customer adoption.
+* **Enhanced Flexibility and Control:** A client-side agent construct gives the application developer full control over the agent's execution loop, state, and tool implementation. This is critical for complex, mission-critical applications where logic cannot be a black box on a server.
+* **Clear Separation of Concerns:** The Llama Stack server is responsible for what it does best: executing LLM inference with tool-calling capabilities. The client is responsible for the application-level concern of orchestrating the agentic logic. This is a clean and scalable architectural pattern.
+* **Stateless Backend:** A stateless API is inherently more scalable, resilient, and easier to manage from an operations perspective. It eliminates concerns about server-side session memory, replication, and expiry.
+
+### Negative
+
+* **Implementation Overhead:** We assume the responsibility for implementing or integrating the client-side Agent construct. This requires dedicated engineering resources, whether for building, testing, or vetting and supporting a third-party solution.
```

Comment:
What do you mean by third-party here? is this ADR assuming its not llama stack where you propose that agent abstraction?